### PR TITLE
Bump SHA-3 accelerator for tutorial enhancements

### PR DESCRIPTION
**Pull request by a0u**
* ucb-bar/sha3#14: Conform to the finalized FIPS 202 standard.
* ucb-bar/sha3#15: Fix virtual memory interaction with interrupts.

This was tested on FireSim.

### CODE REVIEW

File: generators/firechip/src/main/scala/TargetConfigs.scala [hunk 1]
```diff
@@ -138,6 +138,11 @@ class FireSimRocketChipOctaCoreConfig extends Config(
   new WithNDuplicatedRocketCores(8) ++
   new FireSimRocketChipSingleCoreConfig)
 
+// SHA3 accelerator
+class FireSimRocketChipSha3Config extends Config(
```
Comment (alonamid) [inline generators/firechip/src/main/scala/TargetConfigs.scala]: what about the L2?

File: .gitmodules [hunk 2]
```diff
@@ -79,7 +79,7 @@
 	url = https://github.com/freechipsproject/treadle.git
 [submodule "generators/rocc-template"]
```
Comment (abejgonzalez) [inline .gitmodules]:
```suggestion
[submodule "generators/sha3"]
```

**Comment by a0u** [comment]
I made this PR in the meeting before the L2 was discussed.  Time for a rebase.

**Comment by a0u** [comment]
I'm a little ambivalent about having a random `FireSimRocketChipSha3L2Config` in mainline Chipyard.  Maybe this should go into a tutorial-only branch instead of `dev`, but it could be useful to keep around to reference in other educational materials.

Maybe @sagark or @davidbiancolin have an opinion.

**Comment by alonamid** [comment]
Are we also adding the rocket-boom firechip config in this PR?

**Comment by a0u** [comment]
I assumed that would be a separate PR once @abejgonzalez confirms that the heterogeneous config still works.

**Pull request merged**